### PR TITLE
Fixed #1112: Make traces and substances causing allergies or intolerances clickable

### LINKED ISSUE #1112: Make traces and substances causing allergies or intolerances clickable
One of the killer things on Open Food Facts web version is being able to browse all products by clicking on the fields on the product page
It would be great to make the traces field, and the substances causing allergies or intolerances fields  clickable
https://world.openfoodfacts.org/product/3560070575855/tofu-a-l-indienne-riz-basmati-et-lentilles-bio-carrefour-bio

https://world.openfoodfacts.org/trace/eggs
https://world.openfoodfacts.org/trace/eggs.json

https://world.openfoodfacts.org/allergen/soybeans
https://world.openfoodfacts.org/allergen/soybeans.json

### PR DESCRIPTION
**Pull request by l-o-b-s-t-e-r**
Due to lack of progress I've resolved this issues #1112, #1114.

- made allergens, traces and stores clickable
- @teolemon if you don't mind I changed the color of all clickable items, from red to blue.

### CODE REVIEW

**Comment by ManavPoddar** [comment]
@l-o-b-s-t-e-r , you should ask before working on the issues, other contributors are already working on. This wastes the time and hard work we have put in to resolve them. I had already written all the code for #1114 and was waiting for #1161 otherwise there would be merge conflicts.

**Comment by PrajwalM2212** [comment]
@l-o-b-s-t-e-r  https://github.com/openfoodfacts/openfoodfacts-androidapp/pull/1161 . 
I am waiting for the above  pull request to be reviewed and merged . I am doing clickable types for categories , emb and labels.
So please do not work on labels , emb and categories .

**Comment by l-o-b-s-t-e-r** [comment]
@ManavPoddar 
@PrajwalM2212 
Sure, I will close this PR if you have already done this. I just opened the issues and I saw that there is no activity during the last 4-6 days, there is no mentions, so it means that nobody work above it.
Could you please mention in you PR which issues you resolved?

**Comment by Karljoones** [comment]
Hey @PrajwalM2212 and @l-o-b-s-t-e-r #1161 is merged, so if this handled this, we should get this PR. If it does not handle what this PR handles, maybe we can update it to use this new method of handling clickables?

**Comment by PrajwalM2212** [comment]
@Karljoones Yes , this PR removes unnecessary boilerplate code . We can update #1161 to use the method used in this PR.

**Comment by Karljoones** [comment]
Great, can this PR be updated to use the new common method of handling this?

**Comment by l-o-b-s-t-e-r** [comment]
@Karljoones sure)

**Comment by PrajwalM2212** [comment]
@l-o-b-s-t-e-r  please try to not alter the common method we have agreed upon .

**Comment by l-o-b-s-t-e-r** [comment]
@PrajwalM2212 what exactly method do you mean?

**Comment by PrajwalM2212** [comment]
@l-o-b-s-t-e-r  There is a productlistactivity now to handle all searches . I was asking you to not change  modify much in that activity  . There are people working and modifying the activity will lead to conflicts .
If you want to modify the activity you can do so after we have implemented all clickable types.

**Comment by l-o-b-s-t-e-r** [comment]
@Karljoones 
Sorry for delay...

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/utils/SearchType.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+public @interface SearchType {
+
+    String ADDITIVE = "additive";
+    String ALLERGEN = "allergen";
+    String BRAND = "brand";
+    String CATEGORY = "category";
+    String COUNTRY = "country";
+    String EMB = "emb";
+    String LABEL = "label";
+    String PACKAGING = "packagin";
```
Comment (teolemon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/utils/SearchType.java]: packaging

Comment (l-o-b-s-t-e-r) [reply app/src/main/java/openfoodfacts/github/scrachx/openfood/utils/SearchType.java]: @teolemon 
yeah, it's typo, but it doesn't have any effect on the code or appearance
I'll fix

**Comment by Karljoones** [comment]
Code looks good, think it's ready to be merged, but before we do, @l-o-b-s-t-e-r any chance of a screenshot of the final product? On mobile and can't run the code but noticed that you did change the color of the text links

**Comment by l-o-b-s-t-e-r** [comment]
I set the same color as for URL of Product, but if you prefer the red, I can return back.
http://prntscr.com/iqud3t

**Comment by Karljoones** [comment]
I really like that! I think it looks way less intrusive than the red that it was previously!

**Comment by teolemon** [comment]
Merged. Thanks @l-o-b-s-t-e-r @Karljoones :)

**Pull request merged**